scripts/tool_shed/build_ts_whoosh_index.py

```diff
@@ -2,52 +2,81 @@
 Build indexes for searching the Tool Shed.
 Run this script from the root folder, example:
 
-$ python scripts/tool_shed/build_ts_whoosh_index.py -c config/tool_shed.ini --hgweb var/hgweb_config_dir/
+$ python scripts/tool_shed/build_ts_whoosh_index.py -c config/tool_shed.yml
 
-Make sure you adjusted your config to:
- * turn on searching via toolshed_search_on
- * specify whoosh_index_dir where the indexes will be placed
+Make sure you adjusted your Toolshed config to:
+ * turn on searching with "toolshed_search_on"
+ * specify "whoosh_index_dir" where the indexes will be placed
 
 This script expects the Tool Shed's runtime virtualenv to be active.
 """
 from __future__ import print_function
 
+import argparse
 import logging
 import os
 import shutil
 import sys
 import tempfile
 from distutils.dir_util import copy_tree
-from optparse import OptionParser
 
 from mercurial import hg, ui
-from six.moves import configparser
 from whoosh.filedb.filestore import FileStorage
 
 sys.path.insert(1, os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, 'lib')))
 
-import galaxy.webapps.tool_shed.model.mapping
+import galaxy.webapps.tool_shed.model.mapping as ts_mapping
 from galaxy.tool_util.loader_directory import load_tool_elements_from_path
 from galaxy.util import (
     directory_hash_id,
+    ExecutionTimer,
     pretty_print_time_interval,
     unicodify
 )
-from galaxy.webapps.tool_shed import config, model
+from galaxy.util.script import app_properties_from_args, populate_config_args
+from galaxy.webapps.tool_shed import config as ts_config
+from galaxy.webapps.tool_shed import model as ts_model
 from galaxy.webapps.tool_shed.search.repo_search import schema as repo_schema
 from galaxy.webapps.tool_shed.search.tool_search import schema as tool_schema
 from galaxy.webapps.tool_shed.util.hgweb_config import HgWebConfigManager
 
 if sys.version_info > (3,):
     long = int
 
-logging.basicConfig(level='DEBUG')
-
-
-def build_index(sa_session, whoosh_index_dir, path_to_repositories, hgweb_config_dir):
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
```

Comment:
Should this be conditional on `-d/--debug`?

Reply:
Done, thanks!